### ISSUE TITLE
Rethink data loading and validation

### ISSUE THREAD

**Issue by alecgunny** [ML4GW/BBHNet#137]
Especially in the context of #136 , it might make sense to start stripping back some of the omnibus functionality of the dataloaders into pretty simple array-iteration objects, with the option to `shuffle` that iteration. Importantly, rather than loading in the data itself, the data would get loaded in e.g. the `train` project then passed as arrays to the data loader.

The validation dataloader could then look just like the training dataloader, except that the array it used would be a concatenation of the background data pre-batched with copies of that data with the modules outlined in #136 applied to them first (and of course it wouldn't shuffle how it iterated through this data). The dataloader could include logic in the `__init__` about whether the dimension of the data it was passed was 2 or 3 (3 indicating pre-batched data). This will also have the benefit of providing a smoother transition to out-of-memory dataloaders should we ever need to build them.
